### PR TITLE
Remove aten::binary_cross_entropy_with_logits from ATen Fallback

### PR DESCRIPTION
**Pull request by iK1D**
PyTorch removed "binary_cross_entropy_with_logits_backward" from its native function list so that we cannot fallback aten::binary_cross_entropy_with_logits to ATen since we cannot build the gradient graph using that function easily. We have exporter logic for "binary_cross_entropy_with_logits ", when swtiched back to use the original exporter logic, it still works well for both forward and backward.

The ATen fallback for this function is from this PR: https://github.com/microsoft/onnxruntime/pull/9670. At that time we saw customer issue when using original exporter logic. Not sure if this issue is still valid for the customer, if yes, we need to revisit this, fix it, either by changing model script, or exporter logic. @zhijxu-MS

### CODE REVIEW

**Review by baijumeswani — APPROVED** [review]
Looks good.
